### PR TITLE
Add cli arg to override user directory

### PR DESCRIPTION
**Pull request by robinjhuang**
`python main.py --user-directory /path/to/directory`

### CODE REVIEW

File: folder_paths.py [hunk 1]
```diff
@@ -138,6 +146,13 @@ def add_model_folder_path(folder_name: str, full_folder_path: str) -> None:
     else:
         folder_names_and_paths[folder_name] = ([full_folder_path], set())
 
+def replace_folder_path(folder_name: str, full_folder_path: str) -> None:
```
Comment (mcmonkey4eva) [inline folder_paths.py]: what's this for?

File: app/user_manager.py [hunk 2]
```diff
@@ -81,7 +83,7 @@ def add_user(self, name):
         self.users[user_id] = name
 
         global users_file
```
Comment (mcmonkey4eva) [inline app/user_manager.py]: this can be removed, probably should check for other stray globals in this file

File: folder_paths.py (hunk 1 above)
Comment (robinjhuang) [reply folder_paths.py]: Removed

File: app/user_manager.py (hunk 2 above)
Comment (robinjhuang) [reply app/user_manager.py]: Good catch. Removed it and other references in this file with the getter function `get_users_file`.

File: folder_paths.py [hunk 3]
```diff
@@ -78,6 +78,14 @@ def get_input_directory() -> str:
     global input_directory
     return input_directory
 
+def get_user_directory() -> str:
+    global user_directory
```
Comment (mcmonkey4eva) [inline folder_paths.py]: https://stackoverflow.com/questions/11867359/when-do-i-need-to-use-the-global-keyword-in-python

Comment (robinjhuang) [reply folder_paths.py]: Thanks! Removed global.

**Review by mcmonkey4eva — APPROVED** [review]

**Pull request merged**